addons/event_sale/models/event_registration.py

```diff
@@ -48,6 +49,26 @@ def create(self, vals):
                 subtype_id=self.env.ref('mail.mt_note').id)
         return res
 
+    def write(self, values):
+        if 'event_ticket_id' in values:
+            new_ticket_id = self.env['event.event.ticket'].browse(values['event_ticket_id'])
+            if self.event_ticket_id != new_ticket_id:
+                self._sale_order_ticket_type_change_notify(self, new_ticket_id)
+
+        result = super(EventRegistration, self).write(values)
+        return result
+
+    def _sale_order_ticket_type_change_notify(self, event_registration, new_ticket_id):
```

Comment:
I don't really understand, people change ticket on a registration and we trigger an activity on the SO, but don't do anything else ? What kind of bad behavior is that ?

Reply:
The idea here is only to notify the responsible. He will then decide himself what to do with the sale order.

Reply:
Ok